app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt

```diff
@@ -506,6 +541,21 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
 
     private fun setupToolbar() {
         if (Constants.WIDECHAT) {
+
+            if (ShareHandler.hasShare()){
+                with((activity as AppCompatActivity?)){
+                    this?.toolbar?.title = getString(R.string.share_with)
+                    this?.toolbar?.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp)
+                    this?.toolbar?.navigationContentDescription = getString(R.string.go_back_button_description)
+                    this?.toolbar?.setNavigationOnClickListener {
+                        ShareHandler.clear()
+                        setupToolbar()
+                        invalidateOptionsMenu()
```

Comment:
Can we add the search functionality to the toolbar here?  WhatsApp has it, and it will be very useful for when a user has a long list of rooms to scroll through...